x/orderbook/handler.go

```diff
@@ -0,0 +1,100 @@
+package orderbook
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/x"
+)
+
+const (
+	packageName            = "orderbook"
+	newOrderBookCost int64 = 100
+)
+
+// RegisterQuery registers exchange buckets for querying.
+func RegisterQuery(qr weave.QueryRouter) {
+	NewMarketBucket().Register("markets", qr)
+	NewOrderBookBucket().Register("orderbooks", qr)
+}
+
+// RegisterRoutes registers handlers for orderbook message processing.
+func RegisterRoutes(r weave.Registry, auth x.Authenticator, issuer weave.Address) {
+	r = migration.SchemaMigratingRegistry(packageName, r)
+
+	//	r.Handle(CreateOrderBookMsg.Path(), )
+
+	// We do NOT register the TextResultionHandler here... this is only for the proposal Executor
+}
+
+// ------------------- ORDERBOOK HANDLER -------------------
+
+// OrderBookHandler will handle creating orderbooks
+type OrderBookHandler struct {
+	auth            x.Authenticator
+	issuer          weave.Address
+	orderBookBucket *OrderBookBucket
+	marketBucket    *MarketBucket
+}
+
+var _ weave.Handler = OrderBookHandler{}
+
+// NewOrderBookHandler creates a handler that allows issuer to
+// create orderbooks. Only owner/admin of the market can issue
+// new orderbooks
+func (h *OrderBookHandler) NewOrderBookHandler(auth x.Authenticator, issuer weave.Address) weave.Handler {
```

Comment:
This is fine for now, but... isn't issuer based on the Market? And not defined globally in the code (but rather in the data store)?

Creating an orderbook must include the market id, then you can query for the market.Owner